c_common/models/live_packet_gather/src/live_packet_gather.c

```diff
@@ -33,12 +33,16 @@
 #include <eieio.h>
 
 //! Provenance data store
-struct provenance_data_struct {
+typedef struct provenance_data_struct {
     //! Count of overflows when no payload was sent
     uint32_t number_of_overflows_no_payload;
     //! Count of overflows when a payload was sent
     uint32_t number_of_overflows_with_payload;
-};
+    //! Number of events gathered and recorded
+    uint32_t number_of_gathered_events;
+    //! Number of messages sent to host
+    uint32_t number_of_sent_messages;
+} provenance_data_struct;
```

Comment:
our naming pattern for typdedef to always end the name with _t

Reply:
Experience with doxygen suggests:
1. Make the type name unique in the codebase (because otherwise you get weird crosslink errors).
2. Make a typedef with the same name as the struct (because otherwise you will get very confused).

I've gone with that, and I've used the `_t` suffix instead of the ugly `_struct`.